pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -114,6 +114,16 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if kc.TopologyManagerPolicy != kubeletconfig.NoneTopologyManagerPolicy && !localFeatureGate.Enabled(features.TopologyManager) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: TopologyManager %v requires feature gate TopologyManager", kc.TopologyManagerPolicy))
 	}
+	for k, v := range kc.KubeReserved {
+		if v[0] == '-' {
```

Comment:
It feels a little weird to me that we are doing a string equality check to find out if a numeric value is negative... but maybe this is the most sensible way to do this.

Do you know if the values for `KubeReserved` are ever converted to pure integers on which we could do integer comparison checks?

Reply:
I was thinking of doing a string to integer converstion, but I thought that might have been a little overkill, but I have followed @RobertKrawitz suggestion to use the ParseQuantity function so it should be much safer now.